include/PreferenceList.h

```diff
@@ -56,6 +59,17 @@ class PreferenceList {
     /// get pref list element at given index
     [[nodiscard]] PrefListElement at(SizeType index) const;
 
+    //set the ties at the given rank 
+    void set_ties(RankType rank, PreferenceList p); 
+
+    //get the tied elements at the given rank 
+    [[nodiscard]] std::vector<PrefListElement> get_ties(RankType rank) const; 
```

Comment:
This returns a copy of std::vector<PrefListElement>. A const reference to this vector should be returned.
No need for nodiscard here.